library/psa_crypto.c

```diff
@@ -5076,6 +5119,10 @@ static int is_kdf_alg_supported( psa_algorithm_t kdf_alg )
 #if defined(MBEDTLS_PSA_BUILTIN_ALG_TLS12_PSK_TO_MS)
     if( PSA_ALG_IS_TLS12_PSK_TO_MS( kdf_alg ) )
         return( 1 );
+#endif
+#if defined(MBEDTLS_PSA_BUILTIN_ALG_TLS12_ECJPAKE_TO_PMS)
+    if( kdf_alg == PSA_ALG_TLS12_ECJPAKE_TO_PMS )
```

Comment:
It was a request to remove it (it was here), as there's no extraction/transformation of data to be done - the macro consists of just one value, not base + hash.

Reply:
If was surprised by this some time ago too, but apparently the policy is that we only have `IS` macros for algorithms that are also built with a macro, as opposed to being a single value.

For example, `PSA_ALG_RSA_PKCS1V15_CRYPT` is a fixed value, so it doesn't have a corresponding `IS` macro, while the other RSA algs (`PKCS1V15_SIGN`, `PSS`, `OAEP`) are built with a macro as they are parametrized on a hash, so each of them has an `IS` macro.

Reply:
If the algorithm is single-valued, then it is possible to use a direct comparison, `alg == PSA_ALG_XYZ`.

For parameterised algorithms, such an equivalence test is not possible, so to avoid clients building in knowledge of the encoding, the API provides the `PSA_ALG_IS_XYZ(alg)` macros to perform that kind of test.

Not providing the IS macros for single-value algorithms keeps the API definition smaller.